sklearn/metrics/scorer.py

```diff
@@ -67,6 +91,62 @@ def __repr__(self):
                    "" if self._sign > 0 else ", greater_is_better=False",
                    self._factory_args(), kwargs_string))
 
+    def _get_labels_from_estimator(self, estimator, target_type):
+        """ Updates kwargs with ``self._pass_classes``
+
+        Checks if estimator has a classes argument and if scorer has a
+        ``self._pass_classes`` argument. If both exist, it return new kwargs
+        with ``self._pass_classes`` set as the classes_ attrbiute of estimator.
+        This is done only if ``self._pass_classes`` was not already set in
+        ``make_scorer`` initialization
+
+        Parameters
+        ----------
+        estimator : sklearn style estimator
+            The estimator from which classes_ attribute is to be extracted
+
+        target_type : string
+            The type of target y. Should be obtained using ``type_of_target``
+            from ``sklearn.utils.multiclass``
+        """
+
+        if self._pass_classes is None:
+            return self._kwargs
+
+        classes = getattr(estimator, "classes_", None)
+
+        # if classes_ parameter doesn't exist
+        if classes is None:
+            return self._kwargs
+
+        if self._pass_classes in signature(self._score_func).parameters:
+            # if labels passed as kwargs, return kwargs as is
+            if self._pass_classes in self._kwargs:
+                if classes is not None:
+                    # labels should be a subset of classes
+                    if not set(classes).issuperset(
+                                        set(self._kwargs[self._pass_classes])):
+                        raise ValueError("%s classes=%s is not the "
```

Comment:
I'm not convinced we should be stopping the user if their explicitly passed labels include more than the training data. That can be a valid, if unfortunate, case for macro-averaging.

We can warn, or we can be silent, but please no error.

Reply:
yeah agree. and this check should probably be in the specific metric and not here.

but maybe we can a similar check later when "classes" is added to the init of all estimators. Then if the user explicitly specifies classes, labels should be a subset.